src/Document/VectorizedDocument.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Document;
+
+use Symfony\Component\Uid\Uuid;
+use Symfony\Component\Uid\UuidV4;
+
+final class VectorizedDocument extends Document
+{
+    public function __construct(
+        public Uuid $id,
+        public ?string $text,
+        public ?Vector $vector,
```

Comment:
having this nullable feels wrong with the name. and the `hasVector` could be replaced in user land code with `$document instanceof VectorizedDocument`

Reply:
Indeed 👍